src/Handlers/AbstractHandler.php

```diff
@@ -285,11 +288,14 @@ protected function initializeAgents(): void
    * @return string
    * @throws HandlerException
    */
-  protected function addAction(string $hook, $callback, int $priority = 10, int $arguments = 1): string
-  {
-    if (!$this->isValidCallback($callback)) {
+  protected function addAction(string         $hook,
+                               string|Closure $callback,
+                               int            $priority = 10,
+                               int            $arguments = 1
+  ): string {
+    if (!$this->isValidCallback($callback, $arguments)) {
       throw new HandlerException(
-        $this->getInvalidCallbackMessage($callback),
+        'Callback expects more arguments than ' . $arguments . '.',

```

Comment:
the `getInvalidCallbackMessage` was removed because now we throw HanderExceptions that contain context-specific error messages for us.